src/modules/commander/Arming/PreFlightCheck/checks/preArmCheck.cpp

```diff
@@ -134,6 +134,13 @@ bool PreFlightCheck::preArmCheck(orb_advert_t *mavlink_log_pub, const vehicle_st
 		}
 	}
 
+	if (status.is_vtol && status.vehicle_type != vehicle_status_s::VEHICLE_TYPE_ROTARY_WING) {
```

Comment:
Merge with the check above so we can have a single contained vtol section?

Reply:
@sfuhrer Could you please address the review comment so we can get this in?

Reply:
I'd prefer to keep the cases "in transition mode" and "in fixed-wing mode" separated, because the first one should always take place, while the second one can be disabled with a circuit breaker.

Reply:
It's minor, but the point was to only check `status.is_vtol` once, then have the separate "in transition mode" and "in fixed-wing mode" checks within.